### DISCUSSION TITLE
Bypass Package Management - Still use Autoloading

### DISCUSSION THREAD

**Discussion by NRogersBGL** [composer/composer#10782] [Q&A / Support] [answered]
Hi all,

I have this annoying restriction that has been put on our codebase that prevents us from using `composer install/update` to pull in new packages, so I am forced to manually copy in the files needed into the vendor directory. (This is very much not best practice, just required given my specific situation)

I was wondering if there's a way to use composer's great autoloading feature without using composer to pull in the packages.

So to demonstrate:

1. Create a new project
2. Manually create a vendor folder
3. Add code to the vendor folder (mimicking the composer install process)
4. ??? (generate the autoloaded classes/files
5. My project files can then require vendor/autoload.php as normal and no one has to know composer didn't pull in the packages.

Is it at all possible to do this?

Currently I'm working on a project that is rather large and internally they use composer for their deps. Now to extend their codebase with my specific deps ON TOP of what they have, ideally I would just add some composer require statements and be done, HOWEVER (here's the fun part...)
I don't have access to the proxy where their internal code lives.

I have currently requested access or some sort of solution to allow customisations with composer, but in the meantime, I have my project that has some custom composer requirements that I can't load, due to composer install/update failing.

What would be the "best" way to get around this issue and allow for my classes to load. I would really like to not have to manually add requires/includes to all the scripts in my project that reference the dependencies (like we did before autoloading)

If any one has any suggestions or my situation is unclear, please get in touch :)

**Comment by NRogersBGL** [comment]
Knowing how well composer's been built, there probably isa CLI command that does exactly that, and I've just missed it in the list of commands. I'll check again

**Comment by NRogersBGL** [comment, marked as answer]
I think one good workaround could be to copy the files that are currently only accessible from the proxy, to a place locally and use composer to pull them in using the "files" method

**Reply by Seldaek** [reply]
Yes IMO using an artifact repo is the better option here - this can run completely offline if you also disable the packagist.org repo. See https://getcomposer.org/doc/05-repositories.md#artifact and https://getcomposer.org/doc/05-repositories.md#disabling-packagist-org

So you'd download zip archives of the packages you need, put them all in one directory, and then run composer install from that. The only thing you likely need to do is modify every package archive to ensure the composer.json in it has a "version" field defined with the appropriate version. That's a bit of a pain but might be automatable too.. then again if you automate all this you are ultimately doing what Composer does, and one really wonders what kind of nonsense these enterprise gods are imposing on you.

**Reply by NRogersBGL** [reply]
Yeah the solution I ended up going with was removing the external (unaccessible) proxy repo. Instead taking the code that came from that repo and storing it in the local file system, and used the `"type": "path"` repository option. 

It's not the nicest solution, but it works to get around the Enterprise Gods impositions!
